docs/hugo/content/guide/authentication/credential-format.md

```diff
@@ -779,5 +779,76 @@ spec:
+authentication. As a 1st party Microsoft application, one has access to pull a user assigned managed identity's backing
+certificate information from the MSI data plane. Using this data, a user can authenticate to Azure Cloud.
+
+### Prerequisites
+A JSON file with information from the user assigned managed identity. It should be in this format:
+```json
+        {
+            "client_id": "0998...",
+            "client_secret": "MIIKUA...",
+            "client_secret_url": "https://control...",
+            "tenant_id": "93b...",
+            "object_id": "ae...",
+            "resource_id": "/subscriptions/...",
+            "authentication_endpoint": "https://login.microsoftonline.com/",
+            "mtls_authentication_endpoint": "https://login.microsoftonline.com/",
+            "not_before": "2025-02-07T13:29:00Z",
+            "not_after": "2025-05-08T13:29:00Z",
+            "renew_after": "2025-03-25T13:29:00Z",
+            "cannot_renew_after": "2025-08-06T13:29:00Z"
+        }
+```
+
+Note, the client secret should be a base64 encoded certificate.
+
+The steps to get this information from the MSI data plane are as follows:
+1. Make an unauthenticated GET or POST (no Authorization request headers) on the x-ms-identity-url received from ARM to get the token authority and, on older api versions, resource.
+2. Get an Access Token from Azure AD using your Resource Provider applicationId and Certificate. The applicationId should match the one you added to your manifest. The response should give you an access token.
+3. Perform a GET or POST to MSI on the same URL from earlier to get the Credentials using this bearer token.
+
+The only required environment variable is AZURE_USER_ASSIGNED_IDENTITY_CREDENTIALS.
+```bash
+export AZURE_USER_ASSIGNED_IDENTITY_CREDENTIALS="/path/to/credentials-file.json" # The file path to the msi data plane credentials in a JSON file format.
+```
+
+The cloud configuration defaults to Azure Public. If you want to configure the cloud configuration, you will need to 
+export at least AZURE_AUTHORITY_HOST. Both AZURE_RESOURCE_MANAGER_AUDIENCE and AZURE_RESOURCE_MANAGER_ENDPOINT must be 
+specified if you wish to set values for these two fields.
+(see https://learn.microsoft.com/en-us/cli/azure/manage-clouds-azure-cli#list-available-clouds for more info):
+```bash
+export AZURE_AUTHORITY_HOST="https://login.microsoftonline.com/" # The URL of the Entra authority.
+export AZURE_RESOURCE_MANAGER_AUDIENCE=""                        # The Azure Resource Manager Entra audience.
+export AZURE_RESOURCE_MANAGER_ENDPOINT=""                        # The Azure Resource Manager endpoint.
+```
+
+Create or update the `aso-controller-settings` secret:
+
+```bash
+cat <<EOF | kubectl apply -f -
+apiVersion: v1
+kind: Secret
+metadata:
+ name: aso-controller-settings
+ namespace: azureserviceoperator-system
+stringData:
+ AZURE_USER_ASSIGNED_IDENTITY_CREDENTIALS: "$AZURE_USER_ASSIGNED_IDENTITY_CREDENTIALS"
```

Comment:
Is it not also required to inject this file into the pod somehow (i.e. as a mount)?

How are you planning on doing that to consume this? I guess modifying the deployment?